src/uk/gov/hmcts/contino/Helm.groovy

```diff
@@ -75,6 +75,33 @@ class Helm {
     }
   }
 
+  def publishToGitIfNotExists(List<String> values) {
+    addRepo()
+    lint(values)
+
+    def credentialsId = this.steps.env.GIT_CREDENTIALS_ID
+    def gitEmailId = this.steps.env.GIT_APP_EMAIL_ID
+
+    def version = this.steps.sh(script: "helm inspect chart ${this.chartLocation}  | grep ^version | cut -d  ':' -f 2", returnStdout: true).trim()
+    this.steps.echo "Version of chart locally is: ${version}"
+
+    this.steps.writeFile file: 'push-helm-charts-to-git.sh', text: this.steps.libraryResource('uk/gov/hmcts/helm/push-helm-charts-to-git.sh')
+    
+    steps.withCredentials([usernamePassword(credentialsId: credentialsId, passwordVariable: 'BEARER_TOKEN', usernameVariable: 'APP_ID')]) {
+      def publishChart = this.steps.sh (
+        script: "chmod +x push-helm-charts-to-git.sh\n" +
+          "    ./push-helm-charts-to-git.sh ${this.chartLocation} ${this.chartName} $credentialsId $gitEmailId $version",
+        returnStatus: true
+      )
```

Comment:
groovy multiline string is more readable
```suggestion
      def publishChart = this.steps.sh (
        """
        chmod +x push-helm-charts-to-git.sh
        ./push-helm-charts-to-git.sh ${this.chartLocation} ${this.chartName} $credentialsId $gitEmailId $version
        """
      )
```
I would probably also use the credential id and gitEmailId from an env variable too

that would be:
```suggestion
      def publishChart = this.steps.sh (
        script: "chmod +x push-helm-charts-to-git.sh\n" +
          "    ./push-helm-charts-to-git.sh ${this.chartLocation} ${this.chartName} $version",
        returnStatus: true
      )
```

Reply:
cosmetic changes are secondary first need to get the functionality working. These can be done later as well. At the moment `steps.withCredentials([usernamePassword`  is not working.